### PR TITLE
Adding settings option into pause menu QoL

### PR DESCRIPTION
**Pull request by Sector03**
https://user-images.githubusercontent.com/61670787/180659245-4f00be7b-83cc-4521-b941-27d529747176.mp4

small QoL i added to be able to change your settings through the pause screen mid-week and standalone songs.

### CODE REVIEW

**Comment by Mastercopper2346** [comment]
Duplicate of Mode) #8899

**Comment by Stilic** [comment]
> Duplicate of #8899

this one is closed

**Review by DillyzThe1 — APPROVED** [review]
With the the way that options were already implemented in Psych Engine, this seems like a good change.
I have nothing negative to say!

**Review by MemeHoovy — APPROVED** [review]

**Comment by Mastercopper2346** [comment]
> this one is closed

Still a duplicate

File: source/options/OptionsState.hx [hunk 1]
```diff
@@ -106,7 +107,16 @@ class OptionsState extends MusicBeatState
 
 		if (controls.BACK) {
 			FlxG.sound.play(Paths.sound('cancelMenu'));
-			MusicBeatState.switchState(new MainMenuState());
+			if (PlayState.instance != null && OptionsState.fromPlayState) //Check if player came from playstate.
```
Comment (Itz-Miles) [inline source/options/OptionsState.hx:110]: This line fixes the bug in the other PR where reentering/exiting optionsState from any state would load the last song played, regardless of whether it was finished or not.

**Review by GO-Miles — APPROVED** [review]

File: source/options/OptionsState.hx (hunk 1 above)
Comment (REALHaydenGaming) [reply source/options/OptionsState.hx:110]: Pretty cool, adding that in

**Comment by CrusherMods** [comment]
> > this one is closed
> 
> Still a duplicate

this is different code though
the other one only works through charting mode
this one works whenever

File: source/options/OptionsState.hx (hunk 1 above)
Comment (Mastercopper2346) [reply source/options/OptionsState.hx:110]: It was already fixed…

**Comment by Mastercopper2346** [comment]
> his is different code though
the other one only works through charting mode
this one works whenever

it was fixed…

**Comment by Stilic** [comment]
> > his is different code though
> > the other one only works through charting mode
> > this one works whenever
> 
> it was fixed…

wdym

**Review by stilic — APPROVED** [review]

**Comment by LilMewo** [comment]
if this made it into the next update i would freak out i've been trying to import this feature from kade for so long

**Comment by CrusherMods** [comment]
> > his is different code though
> > the other one only works through charting mode
> > this one works whenever
> 
> it was fixed…

what was fixed???

**Review by TheKitBoi — APPROVED** [review]

**Comment by antoSKRLOL** [comment]
cool!!

**Review by JorosTheModderPH — APPROVED** [review]

**Review by Parsellol — APPROVED** [review]

**Review by rodney528 — APPROVED** [review]

**Review by aaaproooo — APPROVED** [review]

**Comment by ImaginationSuperHero528** [comment]
> Duplicate of #8899

this is closed

**Review by rodney528 — APPROVED** [review]

**Pull request closed without merging**